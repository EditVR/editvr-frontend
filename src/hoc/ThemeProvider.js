/**
 * @file ThemeProvider.js
 * Exports a HOC ThemeProvider component that attaches children their MUI theme.
 */

import React from 'react';
import { MuiThemeProvider } from '@material-ui/core';

import muiTheme from '../styles/muiTheme';

const ThemeProvider = WrappedComponent => props => (
  <MuiThemeProvider theme={muiTheme}>
    <WrappedComponent {...props} />
  </MuiThemeProvider>
);

export default ThemeProvider;
