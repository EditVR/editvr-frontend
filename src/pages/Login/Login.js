/**
 * @file Login.js
 * Exports a React component that renders EditVR's login.
 */

import React from 'react';

import { LoginForm } from '../../components';
import { ThinLayout } from '../../layouts';

const Login = () => (
  <ThinLayout title="Log into EditVR">
    <LoginForm />
  </ThinLayout>
);

export default Login;
