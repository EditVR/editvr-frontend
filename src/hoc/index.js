/**
 * @file index.js
 * Exports all hoc components.
 */

import ThemeProvider from './ThemeProvider';

export { ThemeProvider };
