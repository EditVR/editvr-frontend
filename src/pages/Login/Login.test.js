/**
 * @file Login.test.js
 * Contains tests for Login.js.
 */

import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import { ThemeProvider } from '../../hoc';
import Login from './Login';

const ThemedLogin = ThemeProvider(Login);

describe('<Login />', () => {
  it('Matches its snapshot', () => {
    const store = configureStore()({
      loadingBar: {
        default: 0
      },
      user: {
        error: null
      }
    });

    expect(
      renderer
        .create(
          <Provider store={store}>
            <Router>
              <ThemedLogin />
            </Router>
          </Provider>
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});
