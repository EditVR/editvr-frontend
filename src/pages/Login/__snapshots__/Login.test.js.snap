// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Matches its snapshot 1`] = `
<div
  className="ThinLayout-wrapper-1"
  id="layout__wrapper"
>
  <div />
  <div
    className="ThinLayout-thinWrapper-2"
    id="layout__thin-wrapper"
  >
    <header
      className="Header-wrapper-3"
    >
      <img
        alt="EditVR logo"
        className="Header-logo-4"
        src="editvr-logo.svg"
      />
    </header>
    <h1
      className="MuiTypography-root-5 MuiTypography-headline-10"
    >
      Log into EditVR
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-31 LoginForm-textField-29"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-40 MuiInputLabel-root-35 MuiInputLabel-formControl-36 MuiInputLabel-animated-39"
          data-shrink={false}
          htmlFor="username"
        >
          Username
          <span
            className="MuiFormLabel-asterisk-44"
          >
             *
          </span>
        </label>
        <div
          className="MuiInput-root-45 MuiInput-formControl-46 MuiInput-underline-49"
        >
          <input
            aria-invalid={false}
            aria-required={true}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-53"
            defaultValue={undefined}
            disabled={false}
            id="username"
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={true}
            rows={undefined}
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-31 LoginForm-textField-29"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-40 MuiInputLabel-root-35 MuiInputLabel-formControl-36 MuiInputLabel-animated-39"
          data-shrink={false}
          htmlFor="password"
        >
          Password
          <span
            className="MuiFormLabel-asterisk-44"
          >
             *
          </span>
        </label>
        <div
          className="MuiInput-root-45 MuiInput-formControl-46 MuiInput-underline-49"
        >
          <input
            aria-invalid={false}
            aria-required={true}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-53 MuiInput-inputType-56"
            defaultValue={undefined}
            disabled={false}
            id="password"
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={true}
            rows={undefined}
            type="password"
            value={undefined}
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-74 MuiButton-root-58 MuiButton-raised-64 MuiButton-raisedPrimary-65 LoginForm-button-30"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="submit"
      >
        <span
          className="MuiButton-label-59"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root-77"
        />
      </button>
      <a
        className="MuiButtonBase-root-74 MuiButton-root-58 MuiButton-raised-64 MuiButton-raisedSecondary-66 LoginForm-button-30"
        href="/login/reset"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-59"
        >
          Forgot Password
        </span>
        <span
          className="MuiTouchRipple-root-77"
        />
      </a>
    </form>
    <footer
      className="Footer-wrapper-84"
    >
      <div
        className="Footer-elements-85"
      >
        <a
          href="https://fourkitchens.com"
        >
          <p
            className="MuiTypography-root-5 MuiTypography-body1-14 Footer-text-87"
            type="body1"
          >
            EditVR is a product of
          </p>
          <img
            alt="Four Kitchens"
            className="Footer-logo-86"
            src="4k-logo-reversed.svg"
          />
        </a>
      </div>
    </footer>
  </div>
</div>
`;
